src/Raven.Server/Documents/Replication/Senders/MigrationReplicationDocumentSender.cs

```diff
@@ -0,0 +1,115 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Diagnostics;
+using System.IO;
+using System.Text;
+using Raven.Client.Documents.Operations.Replication;
+using Raven.Server.Documents.Handlers;
+using Raven.Server.Documents.Replication.Outgoing;
+using Raven.Server.Documents.Replication.ReplicationItems;
+using Raven.Server.Documents.Replication.Stats;
+using Raven.Server.ServerWide.Context;
+using Raven.Server.Utils;
+using Sparrow.Logging;
+using Sparrow.Server;
+using Sparrow.Server.Utils;
+
+namespace Raven.Server.Documents.Replication.Senders
+{
+    public class MigrationReplicationDocumentSender : ReplicationDocumentSenderBase
+    {
+        public const string MigrationTag = "MOVE";
+        public readonly BucketMigrationReplication Destination;
+        public MigrationReplicationDocumentSender(Stream stream, OutgoingMigrationReplicationHandler parent, Logger log) : base(stream, parent, log)
+        {
+            Destination = parent.BucketMigrationNode;
+        }
+
+        protected override IEnumerable<ReplicationBatchItem> GetReplicationItems(DocumentsOperationContext ctx, long etag, ReplicationStats stats, bool caseInsensitiveCounters)
+        {
+            // TODO : temporary, waiting for RavenDB-17760
+            var migrationId = ctx.DocumentDatabase.ShardedDatabaseId;
+            using var helper = new DocumentInfoHelper();
+            foreach (var item in base.GetReplicationItems(ctx, etag, stats, caseInsensitiveCounters))
```

Comment:
This does linear scan for now, until we have the bucket index.
Important that we can do that on a per etag as well, so big buckets can be done in batches